tests/system_tests_log_level_update.py

```diff
@@ -256,56 +256,69 @@ def router(name, connection):
 
     def test_inter_router_protocol_trace(self):
         qd_manager = QdManager(self, self.address)
-        # Turn off trace logging on all connections for Router B.
-        qd_manager.update("org.apache.qpid.dispatch.log", {"enable": "info+"},
-                          name="log/DEFAULT")
 
+        # The router already has trace logging turned on for all connections.
         # Get the connection id of the inter-router connection
         results = qd_manager.query("org.apache.qpid.dispatch.connection")
         conn_id = None
         for result in results:
             if result['role'] == 'inter-router':
                 conn_id = result['identity']
 
-        # Turn on trace logging for the inter-router connection
-        qd_manager.update("org.apache.qpid.dispatch.connection", {"enableProtocolTrace": "true"}, identity=conn_id)
+        # Turn off trace logging for the inter-router connection. This update command is run async by the router
+        # so we need to sleep a bit before the operation is actually completed.
+        qd_manager.update("org.apache.qpid.dispatch.connection", {"enableProtocolTrace": "false"}, identity=conn_id)
+        time.sleep(1)
```

Comment:
Is there a way to either wait for completion of the async operation or possibly repeatedly query for the current state until the change happens? Just asking.

Reply:
Unfortunately  there is not. We could query the connection again to find out but even if that has the correct value of enableProtocolTrace, it does not guarantee that the action actually has executed. See here - https://github.com/apache/qpid-dispatch/blob/main/src/router_core/agent_connection.c#L644